### PR TITLE
easy custom responses

### PR DESCRIPTION
**Pull request by dgreisen**
verifyResponse is now called to generate the response to the verify
route. Default response is identical to old response. It can now be
overridden in the options.

logoutResponse is now called to generate the response to the logout
route. Default response is identical to old response. It can now be
overridden in the options.

this is useful for people who want to send custom information with their login response (such as user object from database), or for people who already have a front-end that expects a specific format from the server.

### CODE REVIEW

**Comment by jbuck** [comment]
This looks great! I'll add some nitpicks in the comments, but nothing major.

Would you like to take a crack at adding tests for these two parameters too?

File: README.md [hunk 1]
```diff
@@ -94,6 +94,15 @@ Documentation
 * `verifierURI` - The URI of the Persona Remote Verification API
   * Default: `https://verifier.login.persona.org/verify`
   * You probably don't want to touch this unless you have a good reason, like testing.
+* `verifyResponse(error, req, res, email)` - function to generate response for verify route
```
Comment (jbuck) [inline README.md]: s/function/Function/

File: README.md [hunk 2]
```diff
@@ -94,6 +94,15 @@ Documentation
...
+  * Default: see _Verify rout_, below, for successess and failure responses
```
Comment (jbuck) [inline README.md]: s/rout/route/

File: index.js [hunk 3]
```diff
@@ -16,7 +34,7 @@ module.exports = function(app, options) {
 
   var personaOpts = {};
   Object.keys(defaultOptions).forEach(function(key) {
-    if (typeof options[key] === "string") {
+    if (typeof options[key] === "string" || typeof options[key] === "function") {
```
Comment (jbuck) [inline index.js]: Hm, since we're mixing Functions and Strings now, maybe `typeof options[key] === typeof defaultOptions[key]` ?

File: README.md [hunk 4]
```diff
@@ -94,6 +94,15 @@ Documentation
...
+  * `error` will be a string suitable for display (the "reason" attribute in the default implementation), if an error occurs
+  * `req, res` are the request and response objects
+  * `email` is a string containing the email, and will exist if there is not an error
+* `logoutResponse(error, req, res)` - function to generate response for logout route
```
Comment (jbuck) [inline README.md]: s/function/Function/

**Comment by dgreisen** [comment]
Fixed the typos and added test

**Comment by jbuck** [comment]
Bah, of course http://personatestuser.org is down when I need it... I'm going to wait until it's back up before merging this.

**Comment by jbuck** [comment]
Alright, merged and published to npm. Thanks again for doing this!

**Comment by dgreisen** [comment]
sure. Nice byproduct of custom response function that I just noticed is
that you can now pass arbitrary data to the back end for tracking and
whatnot in addition to the assertion. might consider mentioning that in the
docs.

**Pull request merged**